File: UVa10190.java
import java.util.Scanner;

class  main{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        while (sc.hasNext()) {
	        int a, b, temp, ans;
	        temp = 0;
	        int[] num = new int[100];
            a = sc.nextInt();
            b = sc.nextInt();
            num[0] = a;
            while (a!=0&&b!=0&&num[temp] != 0 && temp != 99) {
                if (num[temp] % b != 0 && num[temp] != 1) {
                    temp = 99;
                } else {
                    temp++;
                    num[temp] = num[temp - 1] / b;
                }
            }
            if (temp == 99||temp==0) {
                System.out.println("Boring!");
            } else {
            	for (int i = 0; i < temp - 1; i++) {
                    System.out.print(num[i] + " ");
                }
                System.out.print(num[temp - 1] + "\r\n");
            } 
            
        }
    }
}

File: UVA10190.cpp
#include <iostream>
using namespace std;
int main(){
	int a,b;
	while(cin>>a>>b){
		int size=0,arr[100],temp=a;
		bool flag=true;
		if(b==1||b>a){
			flag=false;
		}
		else{
			while(a>=1){
				if(a>b&&a%b!=0)flag=false;
				arr[size]=a/b;
				size++;
				a/=b;
			}
		}
		if(flag){
			cout<<temp<<" ";
			for(int i=0;i<size-2;i++){
				cout<<arr[i]<<" ";
			}
			cout<<arr[size-2];
			cout<<endl;
		}
		else{
			cout<<"Boring!"<<endl;
		}
	}
}
